### PR TITLE
Animation: Added Whoosh-In animation effect

### LINKED ISSUE #3426: Animation Effect: `whoosh-in`
Recreate the built-in AMP Story animation `whoosh-in`.

Should support, via props, the permutations of `whoosh-in-left` and `whoosh-in-right`.

### Default Properties
`duration`: 500
`delay`: 0

### Desired Output
Two page AMP Story example in Storybook.  The first page implements the AMP Story `whoosh-in-left` and `whoosh-in-right` animations and the second page implements our recreated `whoosh-in` animations.  Both should look the same :sunglasses:

### Helpful Sources
- [AMP Story animation docs](https://amp.dev/documentation/guides-and-tutorials/start/visual_story/animating_elements/?format=stories)
- [Animation product specs](https://docs.google.com/document/d/11xtdVNrSa16qg9tmg_ACw6vTD8z8wymhkRSgOqkVCa8/edit#heading=h.m0pv6unnoqe)

### PR DESCRIPTION
**Pull request by mariano-formidable**
## Summary

This PR adds the `Whoosh In` animation effect to our animation library.

## Relevant Technical Choices

I decided to re-create the whoosh-in effect by combining 3 of our existing animation parts: move, fade, and zoom.  It might have been cleaner to simply create a separate set of keyframes, but I think it was fun to compose the effect and it's a good proof-of-concept when we might want to do composition in the future that can't so easily be created with one set of keyframes.

## User-facing changes

None.

## Testing Instructions

1.) Go to our Storybook: "Animations -> Effects -> Whoosh-In"
2.) Flipped between both pages and notice that the whoosh looks good in both 🎉 

Fixes #3426 

## Screenshots

whoosh_in_effect

### CODE REVIEW

**Review by dmmulroy — APPROVED** [review]
Works in storybook 👍

File: assets/src/dashboard/animations/effects/whooshIn/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  return {
+    id,
+    WAAPIAnimation: ({ children, ...args }) =>
+      moveWAAPIAnimation({
+        children: fadeWAAPIAnimation({
+          children: zoomWAAPIAnimation({ children, ...args }),
+          ...args,
+        }),
+        ...args,
+      }),
```
Comment (maxyinger) [inline assets/src/dashboard/animations/effects/whooshIn/index.js]: is there any benefit of using the functional syntax here over jsx?
```
WAAPIAnimation: ({ children, hoistAnimation }) => (
  <MoveWAAPIAnimation hoistAnimation={hoistAnimation}>
    <FadeWAAPIAnimation hoistAnimation={hoistAnimation}>
      <ZoomWAAPIAnimation hoistAnimation={hoistAnimation}>
        {children}
      </ZoomWAAPIAnimation>
    </FadeWAAPIAnimation>
  </MoveWAAPIAnimation>
),
```

If we find ourselves doing this a lot too, can alway export the more terse `<ComposableWrapper />` from `storyAnimation/WAAPIWrapper.js` to accomplish the same thing:

```
<ComposableWrapper 
  /* could always update this to just take array of WAPPIAnimations too */
  animationParts={[
    {WAAPIAnimation: moveWAAPIAnimation},
    {WAAPIAnimation: fadeWAAPIAnimation},
    {WAAPIAnimation: zoomWAAPIAnimation}
  ]}
  hoistAnimation={hoistAnimation}
>
  {children}
</ComposableWrapper>
```

**Review by maxyinger — APPROVED** [review]
lgtm 🎉

File: assets/src/dashboard/animations/effects/whooshIn/index.js (hunk 1 above)
Comment (mariano-formidable) [reply assets/src/dashboard/animations/effects/whooshIn/index.js]: Ooo, I didn't even think about using the jsx syntax, I'll go head and re-write that way, I think it's much more clear what's happening that way.  

And I agree with your composable wrapper concept too if we find our selves doing this a bunch.

**Review by BrittanyIRL — APPROVED** [review]
This is definitely whooshing! Love that these are getting established in storybook first. 

The end placement of the red square is different for custom vs amp though (i only mention it because in your example they are in the same spot) - I tested in chrome. 

Screen Shot 2020-08-10 at 3 52 19 PM
Screen Shot 2020-08-10 at 3 52 25 PM

**Comment by mariano-formidable** [comment]
> This is definitely whooshing! Love that these are getting established in storybook first.
> 
> The end placement of the red square is different for custom vs amp though (i only mention it because in your example they are in the same spot) - I tested in chrome.
> 
> Screen Shot 2020-08-10 at 3 52 19 PM
> 
> Screen Shot 2020-08-10 at 3 52 25 PM

As long as the custom one places things where it should, I'm good with it :D

**Pull request merged**